daq_main_common.py

```diff
@@ -93,11 +95,17 @@ def process_command_file(command_file_name):
 
 
 def process_immediate_commands(frequency):
+  memory_beat = beamline_support.pvGet(daq_lib.heartbeat)
   while (1):
     if (len(immediate_command_list) > 0):
       command = immediate_command_list.pop(0)
       logger.info('immediate command: %s' % command)
       process_input(command)
+    if memory_beat == beamline_support.pvGet(daq_lib.heartbeat):
```

Comment:
is there any reason you couldn't have this in a separate thread or something? it seems unnecessarily tied to a particular function, and while this is unlikely, the checking interval for the `process_immediate_functions()` could be set in such a way that the popup would always trigger.

Reply:
I only did it to minimize the creation of new threads, but if you think that is a non-issue then I can split it out.